app/src/main/java/com/android/wardrobeManager/ui/images/filters/Checker.java

```diff
@@ -0,0 +1,39 @@
+package com.android.wardrobeManager.ui.images.filters;
+
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.graphics.Color;
+import android.graphics.Paint;
+import android.graphics.PorterDuff;
+import android.graphics.PorterDuffXfermode;
+
+import com.android.wardrobeManager.ui.images.DesignFilterManager;
+
+public class Checker implements DesignFilterManager.DesignFilter {
+    int checkerSize = 40;
+    int colorIndex = 0;
+
+    public Bitmap filter(Bitmap bitmap, Bitmap ref, int[] colors) {
+        bitmap = ref.copy(ref.getConfig(), true);
+        Canvas canvas = new Canvas(bitmap);
+        Paint paint = new Paint(Paint.ANTI_ALIAS_FLAG);
+        paint.setXfermode(new PorterDuffXfermode(PorterDuff.Mode.SRC_IN));
+        paint.setColor(colors[0]);
+        canvas.drawRect(0, 0, bitmap.getWidth(), bitmap.getHeight(), paint);
+
+        for (int i = 0; i < bitmap.getHeight(); i += checkerSize) {
+            for (int j = 0; j < bitmap.getWidth(); j += checkerSize) {
+                if ((i / checkerSize) % 2 == (j / checkerSize) % 2) {
+                    if (colors.length == 1) {
+                        paint.setColor(Color.WHITE);
```

Comment:
I noticed that this is not the only time you require a default color. Maybe define default colors called clothing_item_auto_image_default_color1, clothing_item_auto_image_default_color2 , up to however many you need. That way, these colors are not just hard coded and all designs will end up using the same defaults.

Reply:
If I want to save the default clothing item colors into the colors.xml file I would need to pass a context into the Checker class because that is needed to access the Resources. So then I would need to pass a context into the DesignFilterManager class as well making it more complicated, do you want me to do this or use the the coded color?

Reply:
You don't need to pass the context. You can get the application context with WardrobeManager.getApplicationContext() or something like that. Then use that to resolve the resources.